server/pulp/server/webservices/views/search.py

```diff
@@ -226,6 +226,16 @@ def _trim_results(model, results, only):
     min_fields = set(['_id', 'id', '_href'])
     required_fields = set([field for field, val in model._fields.items() if val.required])
     return_fields = set(only) | min_fields | required_fields
+    # remove subfield from return fields names
+    return_fields_list = list(return_fields)
+    for i in range(len(return_fields_list)):
```

Comment:
This would be a really good use case for python's `enumerate` function.

    for i, field in enumerate(return_fields_list):
        if "." in field:
            return_fields_list[i] = field.split(".")[0]